parcels/examples/example_nemo_curvilinear.py

```diff
@@ -101,6 +104,102 @@ def test_nemo_3D_samegrid():
     assert fieldset.U.dataFiles is not fieldset.W.dataFiles
 
 
+def fieldset_nemo_setup():
+    data_path = path.join(path.dirname(__file__), 'NemoNorthSeaORCA025-N006_data/')
+    ufiles = sorted(glob(data_path + 'ORCA*U.nc'))
+    vfiles = sorted(glob(data_path + 'ORCA*V.nc'))
+    wfiles = sorted(glob(data_path + 'ORCA*W.nc'))
+    mesh_mask = data_path + 'coordinates.nc'
+
+    filenames = {'U': {'lon': mesh_mask, 'lat': mesh_mask, 'depth': wfiles[0], 'data': ufiles},
+                 'V': {'lon': mesh_mask, 'lat': mesh_mask, 'depth': wfiles[0], 'data': vfiles},
+                 'W': {'lon': mesh_mask, 'lat': mesh_mask, 'depth': wfiles[0], 'data': wfiles}}
+    variables = {'U': 'uo',
+                 'V': 'vo',
+                 'W': 'wo'}
+    dimensions = {'U': {'lon': 'glamf', 'lat': 'gphif', 'depth': 'depthw', 'time': 'time_counter'},
+                  'V': {'lon': 'glamf', 'lat': 'gphif', 'depth': 'depthw', 'time': 'time_counter'},
+                  'W': {'lon': 'glamf', 'lat': 'gphif', 'depth': 'depthw', 'time': 'time_counter'}}
+
+    return filenames, variables, dimensions
+
+
+def compute_particle_advection(field_set, mode, lonp, latp):
+
+    def periodicBC(particle, fieldSet, time):
+        if particle.lon > 15.0:
+            particle.lon -= 15.0
+        if particle.lon < 0:
+            particle.lon += 15.0
+        if particle.lat > 60.0:
+            particle.lat -= 11.0
+        if particle.lat < 49.0:
+            particle.lat += 11.0
+
+    def OutOfBounds_reinitialisation(particle, fieldset, time):
+        particle.lat = 5.2
+        particle.lon = 52.0 + (-1e-3 + np.random.rand() * 2.0 * 1e-3)
+
+    pset = ParticleSet.from_list(field_set, ptype[mode], lon=lonp, lat=latp)
+    pfile = ParticleFile("nemo_particles", pset, outputdt=delta(days=1))
+    kernels = pset.Kernel(AdvectionRK4) + periodicBC
+    pset.execute(kernels, runtime=delta(days=4), dt=delta(hours=6),
+                 output_file=pfile, recovery={ErrorCode.ErrorOutOfBounds: OutOfBounds_reinitialisation})
+    return pset
+
+
+@pytest.mark.parametrize('mode', ['jit'])  # Only testing jit as scipy is very slow
+def test_nemo_curvilinear_auto_chunking(mode):
```

Comment:
Why are these functions below here and not in the `example_dask_chunk_OCMs.py` file?

Reply:
hm - so you prefer to take them out here ? In the end, I think the `test_nemo_3D` include those tests anyway.

Reply:
removed the functions as they were a carbon-copy of example_dask_chunk_OCMs.py for NEMO indeed. Thanks for that feedback!